Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp

```diff
@@ -113,36 +119,118 @@ namespace AzToolsFramework
         return returnedNode;
     }
 
+    DPEModelNode* DPEModelNode::GetChildFromDomIndex(size_t domIndex)
+    {
+        // search children for node that has the given dom index
+        // These are almost always tiny lists, so using a Map would be overkill and almost certainly slower in practice
+        DPEModelNode* returnedNode = nullptr;
+
+        // search row and column children for the node with the given DOM index
+        for (auto rowIterator = m_rowChildren.begin(), endIterator = m_rowChildren.end(); rowIterator != endIterator; ++rowIterator)
+        {
+            if ((*rowIterator)->GetDomValueIndex() == domIndex)
+            {
+                returnedNode = *rowIterator;
+                break;
+            }
+        }
+        for (auto columnIterator = m_columnChildren.begin(), endIterator = m_columnChildren.end(); columnIterator != endIterator;
+             ++columnIterator)
+        {
+            if ((*columnIterator)->GetDomValueIndex() == domIndex)
+            {
+                returnedNode = *columnIterator;
+                break;
+            }
+        }
```

Comment:
Is it possible for both searches to return a result? If so, the column will stomp the row.

Reply:
it's not possible, but it is wasteful to search both if the first found it. Fixed this with loop conditions, thanks!